osu.Game/Beatmaps/Formats/Decoder.cs

```diff
@@ -63,7 +63,7 @@ public static Decoder<T> GetDecoder<T>(LineBufferedReader stream)
             if (line == null)
                 throw new IOException("Unknown file format (null)");
 
-            var decoder = typedDecoders.Select(d => line.StartsWith(d.Key, StringComparison.InvariantCulture) ? d.Value : null).FirstOrDefault();
+            var decoder = typedDecoders.Where(d => line.StartsWith(d.Key, StringComparison.InvariantCulture)).FirstOrDefault().Value;
```

Comment:
Why is `.Value` there?

Reply:
We're selecting all of the KeyValuePairs where the key matches the file we're parsing and then getting the value of the first pair that matches.

Reply:
The latest change should make that a bit clearer (and hopefully get rid of the weird InspectCode message)